plugins/org.obeonetwork.m2doc/src-gen/org/obeonetwork/m2doc/template/AbstractProvider.java

```diff
@@ -0,0 +1,85 @@
+/**
+ *  Copyright (c) 2016 Obeo. 
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *   
+ *   Contributors:
+ *       Obeo - initial API and implementation
+ */
+package org.obeonetwork.m2doc.template;
+
+import org.eclipse.emf.common.util.EMap;
+
+import org.obeonetwork.m2doc.provider.IProvider;
+
+/**
+ * <!-- begin-user-doc -->
+ * A representation of the model object '<em><b>Abstract Provider</b></em>'.
+ * <!-- end-user-doc -->
+ *
+ * <p>
+ * The following features are supported:
+ * </p>
+ * <ul>
+ *   <li>{@link org.obeonetwork.m2doc.template.AbstractProvider#getOptionValueMap <em>Option Value Map</em>}</li>
+ *   <li>{@link org.obeonetwork.m2doc.template.AbstractProvider#getProvider <em>Provider</em>}</li>
+ * </ul>
+ *
+ * @see org.obeonetwork.m2doc.template.TemplatePackage#getAbstractProvider()
+ * @model abstract="true"
+ * @generated
+ */
+public interface AbstractProvider extends AbstractConstruct {

```

Comment:
The name seems to be a little bit missleading. I'd rather call this concept 'AbstractProviderClient' or with any other name that reflects the concept (it's not a provider per se).